doc/source/development/building_from_source.rst

```diff
@@ -90,6 +84,41 @@ for the shared lib, *e.g.* ``set (GDAL_LIB_OUTPUT_NAME gdal_x64 CACHE STRING ""
 
 Refer to :ref:`using_gdal_in_cmake` for how to use GDAL in a CMake project.
 
+Building on Windows
++++++++++++++++++++
+
+On Windows, one may need to specify generator:
+
+.. code-block:: bash
+
+    cmake -G "Visual Studio 15 2017" ..
+
+
+Building on MacOS
++++++++++++++++++
+
+On MacOS, there are a couple of libraries that do not function properly when the GDAL build requirements are installed using Homebrew.
+
+The `Apache Arrow <https://arrow.apache.org/docs/index.html>`_ library included in the current distribution of CMake is broken, and causes a detection issue. In order to build GDAL successfuly, you must first remove it:
+
+.. code-block:: bash
+
+    rm -rf /usr/local/lib/cmake/Arrow
+
+And then set the appropriate options to exclude Arrow from the build:
+
+.. code-block:: bash
+
+    cmake -DGDAL_USE_ARROW=OFF -DGDAL_USE_PARQUET=OFF ..
+
+
+Similarly, recent versions of Homebrew no longer bundle `Boost <https://www.boost.org/>`_ with libkml, causing a failure to find Boost headers. You should either install Boost manually or disable libkml when building on MacOS:
+
+.. code-block:: bash
+
+    cmake -DGDAL_USE_LIBKML=OFF ..
```

Comment:
Similar: Disable find_package to work around the issue. And proabably not broken in vcpkg.

Reply:
I don't have time/the environment to investigate that, but indeed this is weird. I'm not sure why the build recipee of GDAL in homebrew would work with boost + libkml installed, and not manual GDAL builds. Or maybe because the boost headers are in a non standard location that their build script adds an include for ?? Maybe the authors of the libkml boost change in homebrew could bring some light on this

Reply:
> See discusion here: [Homebrew/homebrew-core#128083](https://github.com/Homebrew/homebrew-core/pull/128083)
> It looks like Boost is being intentionally excluded.

Hm, AFAIU they were concerned about the size of (their!) boost package and decided to focus on runtime usage.

GDAL could update its `FindLibKML.cmake` to ensure boost is available.

> Though from other discussions on [this PR](https://github.com/OSGeo/gdal/pull/7588), it looks like Boost fails even when installed manually, so there may be something else going on. Whether that's an issue with brew, boost, or gdal, I'm unsure.

https://github.com/OSGeo/gdal/pull/7588#issuecomment-1681626135. Not a gdal issue.